ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/load/table/LoadPartitions.java

```diff
@@ -176,11 +173,29 @@ private boolean isMetaDataOp() {
    * Also, copy relevant stats and other information from original request.
    *
    * @throws SemanticException
+   * @param lastAlterTableAddPartitionDesc
    */
-  private void addConsolidatedPartitionDesc() throws Exception {
+  private void addConsolidatedPartitionDesc(AlterTableAddPartitionDesc lastAlterTableAddPartitionDesc) throws Exception {
+    int maxTasks = 0;
     //Load partitions equal to batch size at one go for metadata only and for external tables.
-    int maxTasks = context.hiveConf.getIntVar(HiveConf.ConfVars.REPL_LOAD_PARTITIONS_BATCH_SIZE);
+    if (isMetaDataOp() || TableType.EXTERNAL_TABLE.equals(table.getTableType())) {
```

Comment:
Any reason for doing this? This would break HIVE-23520. Sorry about noticing this late.